localstack/testing/aws/util.py

```diff
@@ -180,7 +181,10 @@ def base_aws_client_factory(session: boto3.Session) -> ClientFactory:
     config = None
     if os.environ.get("TEST_DISABLE_RETRIES_AND_TIMEOUTS"):
         config = botocore.config.Config(
-            connect_timeout=1_000, read_timeout=1_000, retries={"total_max_attempts": 1}
+            connect_timeout=1_000,
+            read_timeout=1_000,
+            retries={"total_max_attempts": 1},
+            region_name=TEST_AWS_REGION_NAME,
```

Comment:
Not sure about this one. I think it might be better if the region is only added as a fallback in the else branch below? (i.e. when `os.environ.get("TEST_TARGET") != "AWS_CLOUD"`)